test/jsonapi/serializer_test.exs

```diff
@@ -536,39 +535,31 @@ defmodule JSONAPI.SerializerTest do
     Application.delete_env(:jsonapi, :remove_links)
   end
 
-  test "serialize includes pagination links if they are defined and with_pagination is configured" do
-    data = %{id: 1}
-    Application.put_env(:jsonapi, :with_pagination, true)
+  test "serialize includes pagination links if page-based pagination is requested" do
+    data = [%{id: 1}]
 
-    encoded = Serializer.serialize(PostView, data, nil)
+    conn =
+      :get
+      |> Plug.Test.conn("/mytype?page[page]=2&page[size]=1")
+      |> QueryParser.call(%Config{view: PaginatedPostView, opts: []})
 
-    assert encoded[:links][:next] ==
-             PostView.url_for_pagination(data, nil, %{cursor: "some-string"})
+    encoded =
+      Serializer.serialize(PaginatedPostView, data, conn, nil, total_pages: 3, total_items: 3)
 
-    Application.delete_env(:jsonapi, :with_pagination)
+    assert encoded[:links][:first]
```

Comment:
Should we consider actually asserting on what we expect the URL to look like?

Reply:
Sure, will update the tests.